ClientPlugin/Settings/Elements/Textbox.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Sandbox.Graphics.GUI;
+using System;
+using System.Collections.Generic;
+
+namespace ClientPlugin.Settings.Elements
+{
+    internal class TextboxAttribute : Attribute, IElement
+    {
+        public readonly string Description;
+
+        public TextboxAttribute(string description = null)
+        {
+            Description = description;
+        }
+
+        public List<MyGuiControlBase> GetElements(string name, Func<object> propertyGetter, Action<object> propertySetter)
+        {
+            var textBox = new MyGuiControlTextbox(defaultText: (string)propertyGetter());
+            textBox.TextChanged += (box)=>propertySetter(box.Text);
+
```

Comment:
```cs
            if (!string.IsNullOrEmpty(Description))
                textBox.SetToolTip(Description);
```

Reply:
Tooltips have been added to textboxes and keybinds
The null check is done as part of SE's code